### PR TITLE
Create the classes

### PR DESCRIPTION
**Pull request by eng-mohamed-eldeeb**
-Create the Person class with is_of_age and can_use_services methods.

-Create a Student class that inherits from the Person class with the play_hooky method.

-Create a teacher class that inherits from the Person class and modify the can_use_services method :)

### CODE REVIEW

File: person_class.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    is_of_age || parent_permission
+  end
```
Comment (Deepakdanger) [inline person_class.rb]: - [x] Kindly use `of_age?` in line 17 as `def of_age?` is defined above.

File: student_class.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require './person_class'
+class Student < Person
+  def initialize(classroom)
+    super()
+    @classroom = classroom
+  end
```
Comment (Deepakdanger) [inline student_class.rb]: - [x] Here at the time of initialization we have to pass four parameters. (age, classroom, name = "Unknown", parent_permission = true). some of them have default values. Kindly use-
```ruby
 def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```

File: teacher_class.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require './person_class'
+class Teacher < Person
+  def initialize(specialization)
+    super()
+    @specialization = specialization
+  end
```
Comment (Deepakdanger) [inline teacher_class.rb]: - [x] Here at the time of initialization we have to pass four parameters. (age, `specialization`, name = "Unknown", parent_permission = true). some of them have default values. Kindly use-
```ruby
 def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @eng-mohamed-eldeeb , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
- [x] README file should be descriptive:

- All main headers from the [README template](https://github.com/microverseinc/readme-template) should be present in a student's repo
- KIndly fill all the necessary details.
- [x] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person_class.rb [hunk 4]
```diff
@@ -14,6 +14,6 @@ def of_age?
   end
 
   def can_use_services?
-    is_of_age || parent_permission
+    of_age || parent_permission
```
Comment (omarsalem7) [inline person_class.rb]: - [ ] As stated by the previous reviewer, Kindly call the correct method which is `of_age?` not `of_age`

File: README.md [hunk 5]
```diff
@@ -1 +1,56 @@
...
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Microverse's Curriculum project
+- Thanks to all code reviewers
+
+## 📝 License
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (omarsalem7) [inline README.md]: - [x] As stated previously by the reviewer, consider using [a new README template](https://github.com/microverseinc/readme-template) not the old one and make sure to include all main headers. Also, consider customizing it for your project.

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @eng-mohamed-eldeeb 😊, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

Almost near

 ## Highlights
✔️ There is no linter error present
✔️ The Pull Request has a proper title and description

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] [OPTIONAL] Kindly consider using the correct gitflow by opening a pull request from `your_branch` to `development` branch not `main` in your next pull request because this is one of the base requirements for the project which it misses by the previous reviewer.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmiiorji — APPROVED** [review]
Hi @eng-mohamed-eldeeb, 

# Status: Approved ✅

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Congrats

## Good Highlights✔️

- Linter checks are passing
- Good commit messages
- Adhered to Gitflow
- Requested changes have been implemented


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**